ha/core/event_analyzer/parser/parser.py

```diff
@@ -176,36 +263,51 @@ def parse_event(self, msg: str) -> HealthEvent:
-            event_id = timestamp + str(uuid.uuid4().hex)
-            node_id = cluster_resource_alert["_resource_name"]
-            resource_type = cluster_resource_alert["_resource_type"]
-            event_type = cluster_resource_alert["_event_type"]
-            timestamp = cluster_resource_alert["_timestamp"]
+            health_event_list = []
+            if self._conf_store is not None:
+                key_to_search = self.prepare_key(cluster_resource_alert)
+                # Log.error(key_to_search)
+                if self._conf_store.key_exists(key_to_search):
+                    sys_health_key_val, current_pod_restart_val, gen_id = \
+                       ClusterResourceParser._get_sys_health_key_val(key_to_search, self._conf_store)
+                    # Log.error(f'{cluster_resource_alert["_generation_id"]} {gen_id}')
+                    if cluster_resource_alert["_generation_id"] != gen_id:
+                        if current_pod_restart_val is not None and current_pod_restart_val:
+                            # If the incoming generation id doesn't match with already stored
+                            # value and assosciated pod_restart count is 1. That means,
+                            # its a pod restart case and this alert is already handled
+                            sys_health_key_val["events"][0]["specific_info"]["pod_restart"] = 0
+                            self._conf_store.update(key_to_search, json.dumps(sys_health_key_val))
+                        else:
+                            # generation id is different and pod_restart value assosciated with
+                            # this pod is also zero. Means, pod restart happened and online event
+                            # came first(before delete event). So, handle two events here itself
 
-            event = {
-                EVENT_ATTRIBUTES.EVENT_ID : event_id,
-                EVENT_ATTRIBUTES.EVENT_TYPE : event_type,
-                EVENT_ATTRIBUTES.SEVERITY : StatusMapper.EVENT_TO_SEVERITY_MAPPING[event_type],
-                EVENT_ATTRIBUTES.SITE_ID : site_id, # TODO: Should be fetched from confstore
-                EVENT_ATTRIBUTES.RACK_ID : rack_id, # TODO: Should be fetched from confstore
-                EVENT_ATTRIBUTES.CLUSTER_ID : cluster_id, # TODO: Should be fetched from confstore
-                EVENT_ATTRIBUTES.STORAGESET_ID : node_id,
-                EVENT_ATTRIBUTES.NODE_ID : node_id,
-                EVENT_ATTRIBUTES.HOST_ID : node_id,
-                EVENT_ATTRIBUTES.RESOURCE_TYPE : resource_type,
-                EVENT_ATTRIBUTES.TIMESTAMP : timestamp,
-                EVENT_ATTRIBUTES.RESOURCE_ID : node_id,
-                EVENT_ATTRIBUTES.SPECIFIC_INFO : "specific_info"
-            }
+                            # One potantial problem in sending two alerts here is if after pod
+                            # comes up on some other node, then failed alert "node" will be wrongly sent.
```

Comment:
Not comment to fix, but for clarity, -
Assumption here is that pod with machineID xyz (/etc/machinid), going down on physical_server_abc, if comes back on physical_server_def will still maintain it's machineID to be xyz, right? Also there is no physical server information going out in event fields, right? Then why "failed alert "node" will be wrongly sent.??"

Reply:
This was in terms of timestamp. The timestamp we will be facing for failed alert will not be correct. But that doesn't affect the functionality and no major impact as such.